src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.cs

```diff
@@ -260,5 +261,239 @@ public static bool HasAllSet(this BitArray bitArray)
+
+            // There is differentiation in the integral and fractional lengths,
+            // concatenate both into singular buffers and compare them.
+            scoped Span<byte> leftDigits;
+            scoped Span<byte> rightDigits;
+            byte[]? rentedLeftBuffer;
+            byte[]? rentedRightBuffer;
+
+            if (nDigits <= JsonConstants.StackallocByteThreshold)
+            {
+                leftDigits = stackalloc byte[JsonConstants.StackallocByteThreshold];
+                rightDigits = stackalloc byte[JsonConstants.StackallocByteThreshold];
+                rentedLeftBuffer = rentedRightBuffer = null;
+            }
+            else
+            {
+                leftDigits = (rentedLeftBuffer = ArrayPool<byte>.Shared.Rent(nDigits));
+                rightDigits = (rentedRightBuffer = ArrayPool<byte>.Shared.Rent(nDigits));
+            }
+
+            leftIntegral.CopyTo(leftDigits);
+            leftFractional.CopyTo(leftDigits.Slice(leftIntegral.Length));
+            rightIntegral.CopyTo(rightDigits);
+            rightFractional.CopyTo(rightDigits.Slice(rightIntegral.Length));
+
+            bool result = leftDigits.Slice(0, nDigits).SequenceEqual(rightDigits.Slice(0, nDigits));
+
+            if (rentedLeftBuffer != null)
+            {
+                Debug.Assert(rentedRightBuffer != null);
+                rentedLeftBuffer.AsSpan(0, nDigits).Clear();
+                rentedRightBuffer.AsSpan(0, nDigits).Clear();
+                ArrayPool<byte>.Shared.Return(rentedLeftBuffer);
+                ArrayPool<byte>.Shared.Return(rentedRightBuffer);
+            }
+
+            return result;
+
+            static void ParseNumber(
+                ReadOnlySpan<byte> span,
+                out bool isNegative,
+                out ReadOnlySpan<byte> integral,
+                out ReadOnlySpan<byte> fractional,
+                out int exponent)
+            {
+                // Parses a JSON number into its integral, fractional, and exponent parts.
+                // The returned components use a normal-form representation wherein two numbers
+                // are equal if and only if the sign and exponent are equal and additionally the
+                // concatenation of the integral and fractional parts are sequence equal.
+                // Under this scheme the number 0 is represented by a pair of empty spans.
+
+                Debug.Assert(span.Length > 0);
+
+                if (span[0] == '-')
+                {
+                    isNegative = true;
+                    span = span.Slice(1);
+                }
+                else
+                {
+                    Debug.Assert(char.IsDigit((char)span[0]), "leading plus not allowed in valid JSON numbers.");
+                    isNegative = false;
+                }
+
+#if NET
+                int i = span.IndexOfAny(s_decimalPointOrExponent);
+#else
+                int i = span.IndexOfAny((byte)'.', (byte)'e', (byte)'E');
```

Comment:
For three inputs like this, I'd just use this version everywhere rather than special-casing for SearchValues; I wouldn't expect SearchValues to be meaningfully better.

Reply:
Removed.